ndk/src/main/jni/utils/stack_unwinder_libunwindstack.cpp

```diff
@@ -0,0 +1,68 @@
+#include "../utils/string.h"
+#include "stack_unwinder_libunwindstack.h"
+#include <stdlib.h>
+#include <ucontext.h>
+#include <unwindstack/Elf.h>
+#include <unwindstack/MapInfo.h>
+#include <unwindstack/Maps.h>
+#include <unwindstack/Memory.h>
+#include <unwindstack/Regs.h>
+
+ssize_t
+bsg_unwind_stack_libunwindstack(bsg_stackframe stacktrace[BUGSNAG_FRAMES_MAX],
+                                siginfo_t *info, void *user_context) {
+  if (user_context == NULL) {
+    return 0; // only handle unwinding from signals
+  }
+
+  const std::unique_ptr<unwindstack::Regs> regs(
+      unwindstack::Regs::CreateFromUcontext(unwindstack::Regs::CurrentArch(),
+                                            user_context));
+
+  // if (user_context != NULL) {
+  // regs = unwindstack::Regs::CreateFromUcontext(
+  // unwindstack::Regs::CurrentArch(), user_context);
+  //} else {
+  // regs = unwindstack::Regs::CreateFromLocal();
+  //}
+
+  std::string unw_function_name;
+  unwindstack::LocalMaps maps;
+
+  if (!maps.Parse()) {
+    stacktrace[0].frame_address = regs->pc(); // only known frame
+    return 1;
+  }
+
+  const std::shared_ptr<unwindstack::Memory> memory(
+      new unwindstack::MemoryLocal);
+
+  int frame_count = 0;
+  for (int i = 0; i < BUGSNAG_FRAMES_MAX; i++) {
+    unwindstack::MapInfo *const map_info = maps.Find(regs->pc());
+    unwindstack::Elf *const elf = map_info->GetElf(memory, false);
+    if (!map_info || !elf) {
```

Comment:
If `map_info` can be null at this point then I believe we'd also need to perform a null check at `map_info->GetElf`?